server/db/model.py

```diff
@@ -83,75 +210,70 @@ def _session(self):
+    def campaigns_list(self):
+        """
+        Get campaign list (last added will be at first), can be empty
+        Returns:
+            list: list of Campaign instances
+        """
+        with no_auto_dereference(Campaign):
+            return list(Campaign.objects.raw({"creator": self._id}))[::-1]
 
-class Prospect(MongoModel):
-    """
-    Prospect Model
-    Example:
-      from pymodm.connection import connect
-      connect("mongodb://localhost:8000/db", alias="user-db")
-      new_prospect=Prospect("steven@example.com", first_name="Steven", last_name="McGrath", status="open").save()
+    def campaigns_append(self, **campaign_info):
+        """
+        Add campaigns for the user
+        Args:
+            **campaign_info: keyword dict that Campaign need
 
-    """
-    owner_email = fields.EmailField()
-    email = fields.EmailField()
-    first_name = fields.CharField()
-    last_name = fields.CharField()
-    status = fields.CharField()
+        Returns:
+            Campaign: Campaign instance
+        """
+        c = Campaign(creator=self._id, creation_date=datetime.now(),
+                     **campaign_info).save()
+        self.campaigns_count += 1
+        self.save()
+        with no_auto_dereference(Campaign):
+            return c
 
-    @staticmethod
-    def get_by_owner_email(owner_email):
+    def campaign_by_id(self, campaign_id):
         """
-        This method finds all prospects which have specified owner_email
+        Get campaign by id
+        Args:
+            campaign_id: id
+        Returns:
+            Campaign: Campaign object
+        Raises:
+            DoesNotExist
         """
-        ret = Prospect.objects.raw({"owner_email": owner_email})
-        ret_list = list(ret)
-        return ret_list if ret_list else None
+        return Campaign.objects.get({"$and": [{"_id": ObjectId(campaign_id)}, {"creator": self._id}]})
 
-    @staticmethod
-    def check_duplicate_prospect(owner_email, email):
+    @property
+    def campaigns(self):
+        with no_auto_dereference(Campaign):
+            return list(Campaign.objects.raw({"creator": self._id}))
+
+    def prospects_bulk_append(self, prospects_list):
         """
-        This method checks whether the specified owner email is
-        already associated with the specified email
+        Append Prospect to User; Prospect is store for each user
+        Args:
+            prospects_list: list of keyword dict to construct Prospect
+        Returns:
+            list: list of id of prospects
         """
-        ret = Prospect.objects.raw(
-            {"owner_email": owner_email, "email": email})
-        ret_list = list(ret)  # might just need to  check ret
-        return True if ret_list else None
+        # TODO: handle repeat/exists prospects
+        prospects_objs = Prospect.objects.bulk_create(
+            [Prospect(owner=self._id, **each_p) for each_p in prospects_list if each_p["email"] not in self.prospects], retrieve=False)
```

Comment:
self.prospects doesn't exist anymore, right @Patton-L - so in this case, I would revert back to how you did it before Steven, but just add the logic to this file

Reply:
Looks like "prospects" is a property on User model. Line 267